chain/chain/src/types.rs

```diff
@@ -127,6 +132,85 @@ impl BlockHeaderInfo {
     }
 }
 
+/// Block economics config taken from genesis config
+pub struct BlockEconomicsConfig {
+    gas_price_adjustment_rate: Rational,
+    min_gas_price: Balance,
+    max_gas_price: Balance,
+    genesis_protocol_version: ProtocolVersion,
+}
+
+impl BlockEconomicsConfig {
+    /// Compute min gas price according to protocol version and chain id. We only upgrade gas price
+    /// for betanet, testnet and mainnet.
+    pub fn min_gas_price(&self, protocol_version: ProtocolVersion) -> Balance {
+        if self.genesis_protocol_version < protocol_version
```

Comment:
`if self.genesis_protocol_version < MIN_PROTOCOL_VERSION_NEP_92`

Reply:
That doesn't work. Let's say `genesis_protocol_version` is 29 and current protocol version is also 29, whereas `MIN_PROTOCOL_VERSION_NEP_92` is 31, then the gas price should be `self.min_gas_price` instead of `MIN_GAS_PRICE_NEP_92`

Reply:
In your example the check on the very next line will fail.